### PR TITLE
Fixes #30507: Use enabled hook helpers for Katello and Foreman

### PR DESCRIPTION
**Pull request by ehelms**

### CODE REVIEW

File: hooks/pre/10-reset_data.rb
Comment (ekohl) [inline hooks/pre/10-reset_data.rb]: Just thinking out loud: do we also want to start to prepare for an optional candlepin and introduce `candlepin_enabled?`?

File: katello/hooks/pre/15-check_java.rb
Comment (ekohl) [inline katello/hooks/pre/15-check_java.rb]: This would also be `candlepin_enabled?`

File: katello/hooks/pre/20-certs_update.rb
Comment (ekohl) [inline katello/hooks/pre/20-certs_update.rb:23]: This part should really be a `pre_commit` hook IMHO. In `pre` the answers are already saved to disk. You should fail on invalid user input in `pre`.

It should also check if the `certs` module is enabled, rather than `katello`, right? It checks if the `certs` params are passed in and I think that `server_ca_cert` can also be used on a content proxy.

File: hooks/pre/10-reset_data.rb
Comment (ehelms) [reply hooks/pre/10-reset_data.rb]: I thought about that with my prior PR but we do not have a clean way to detect that I could think of. So I kept it to a clearer representation of how things are.

File: katello/hooks/pre/20-certs_update.rb
Comment (ehelms) [reply katello/hooks/pre/20-certs_update.rb:23]: File an issue if you don't mind, touching our certs code requires a Friday :P

File: hooks/pre/10-reset_data.rb
Comment (ekohl) [reply hooks/pre/10-reset_data.rb]: Well, for now we just have it equal to `katello_enabled?` but it prepares the rest of the codebase. Once we circle back to https://github.com/theforeman/foreman-installer/pull/421 it will be good.

**Comment by ehelms** [comment]
Added candlepin_enabled? helper

File: katello/hooks/pre_commit/14-cdn_setting.rb
Comment (ekohl) [inline katello/hooks/pre_commit/14-cdn_setting.rb]: I think this doesn't make that much sense. This is very much into the actual config and answers. It's not about a high level concept.

File: hooks/boot/01-kafo-hook-extensions.rb
Comment (ekohl) [inline hooks/boot/01-kafo-hook-extensions.rb]: I think this only makes it less clear how this works. I'd suggest this only reduces the understanding of the flow.

If anything, this should be `candlepin_enabled?`

**Comment by ehelms** [comment]
This could use another round of review

**Review by ekohl — COMMENTED** [review]
Looks like I had comments that I failed to submit

File: katello/hooks/pre_commit/14-cdn_setting.rb
Comment (ehelms) [reply katello/hooks/pre_commit/14-cdn_setting.rb]: I don't follow the request here.

File: katello/hooks/pre/15-check_java.rb
Comment (ekohl) [inline katello/hooks/pre/15-check_java.rb:20]: On an unrelated note: I think this check itself should ensure JDK 1.8 is used since 1.7 is too old.

File: katello/hooks/boot/13-hiera.rb
Comment (ekohl) [inline katello/hooks/boot/13-hiera.rb]: I'm going to hold off on merging this in favor of merging https://github.com/theforeman/foreman-installer/pull/542 first. IMHO that solution is cleaner and you get to deal with the merge conflict.

**Review by ekohl — COMMENTED** [review]
Overall this looks ok, but holding off until we merge tuning.

**Review by ekohl — APPROVED** [review]

**Pull request merged**
